crates/subspace-farmer/src/multi_farming.rs

```diff
@@ -99,18 +116,100 @@ impl MultiFarming {
                             plot.clone(),
                             plot_commitments.clone(),
                             client.clone(),
-                            identity,
+                            identity.clone(),
                             reward_address,
                         )
                     });
 
-                    Ok::<_, anyhow::Error>((plot, subspace_codec, plot_commitments, farming))
+                    Ok::<_, anyhow::Error>((
+                        identity,
+                        plot,
+                        subspace_codec,
+                        plot_commitments,
+                        farming,
+                    ))
+                })
+            })
+            .collect::<FuturesOrdered<_>>()
+            .enumerate();
+
+        let starting_port = if let Some(mut starting_multiaddr) = listen_on.clone() {
```

Comment:
Looks like `map()` refactoring candidate.

Reply:
It's not as I return error below

Reply:
Can't find this code now. Probably, it's outdated.